core/commands/resolve.go

```diff
@@ -78,6 +84,16 @@ Resolve the value of another name recursively:
 			depth = namesys.DefaultDepthLimit
 		}
 
+		if strings.HasPrefix(name, "/ipfs/") || !strings.HasPrefix(name, "/") {
+			resolved, err := resolveIpfsPath(req.Context(), n.Resolver, name)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+			res.SetOutput(&ResolvedPath{resolved})
+			return
+		}
```

Comment:
I thought that would be the expected behavior. Maybe it would make sense to do the former by default, and the latter when the `-r` flag is used?